FrontPage/Client.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+import Apollo
+import AppAuth
+
+let config = Config()
+let syncUrl = config.getConfiguration("sync-app")
+let syncConfig = syncUrl.config ?? ["error": JSONValue.bool(false)]
+let syncWs = syncConfig["websocketUrl"]?.getString() ?? "error";
+
+class Client{
+  static let instance = Client()
+  let auth = AuthViewController()
```

Comment:
so here all we really need is the token value. While the current code does work, it however brings tight coupling between the `Client` class and the `AuthViewController` class, which will make it hard to test the `Client` class (if we are going to add unit tests for example).

A simple way to optimise this is rather than `Client` call to `AuthViewController` to get the token, we can make token a static field on the `Client` class, and then let the `AuthViewController` to assign the `token` value to the `Client` class. Basically it's another way around: the `AuthViewController` will set the `token` value on the `Client` class directly. Similarly if there are other classes need to use the token, they will do the same thing.

This way the `Client` class no longer need to reference `AuthViewController`, and it will be easy to test this class if we need to.